### PR TITLE
Fix NullPointerException when doing knn search on empty index without dims

### LINKED ISSUE #111733: vector search error when the target index is empty
### Elasticsearch Version

8.14.3

### Installed Plugins

_No response_

### Java Version

_bundled_

### OS Version

CentOS 9

### Problem Description

Executing this request results in `400` error.
This is because the index is empty.

```
GET /esre-team-test-chunks/_search
{
  "fields": [
    "text"
  ],
  "knn": {
    "field": "text_embedding",
    "k": 10,
    "num_candidates": 100,
    "query_vector_builder": {
      "text_embedding": {
        "model_id": ".multilingual-e5-small",
        "model_text": "test"
      }
    }
  }
}
```

- response.

```
{
  "error": {
    "root_cause": [
      {
        "type": "query_shard_exception",
        "reason": """failed to create query: Cannot invoke "java.lang.Integer.intValue()" because "this.dims" is null""",
        "index_uuid": "z60wltlVSa-kfCZsAAapOA",
        "index": "esre-team-test-chunks"
      }
    ],
    "type": "search_phase_execution_exception",
    "reason": "all shards failed",
    "phase": "dfs",
    "grouped": true,
    "failed_shards": [
      {
        "shard": 0,
        "index": "esre-team-test-chunks",
        "node": "oTp_Zo49QByHhOB0DDJP_g",
        "reason": {
          "type": "query_shard_exception",
          "reason": """failed to create query: Cannot invoke "java.lang.Integer.intValue()" because "this.dims" is null""",
          "index_uuid": "z60wltlVSa-kfCZsAAapOA",
          "index": "esre-team-test-chunks",
          "caused_by": {
            "type": "null_pointer_exception",
            "reason": """Cannot invoke "java.lang.Integer.intValue()" because "this.dims" is null"""
          }
        }
      }
    ]
  },
  "status": 400
}
```

### Steps to Reproduce

- Please try this script in Dev tools.

```
DELETE esre-team-test-chunks

PUT esre-team-test-chunks
{
  "mappings": {
    "dynamic_templates": [],
    "properties": {
      "metadata": {
        "properties": {
          "esre": {
            "properties": {
              "created_at": {
                "type": "date"
              },
              "embedding_model": {
                "type": "text"
              },
              "file_id": {
                "type": "text"
              },
              "team_id": {
                "type": "text"
              },
              "updated_at": {
                "type": "date"
              }
            }
          },
          "source": {
            "type": "text",
            "fields": {
              "keyword": {
                "type": "keyword",
                "ignore_above": 256
              }
            }
          }
        }
      },
      "text": {
        "type": "text",
        "fields": {
          "ngram": {
            "type": "text"
          }
        }
      },
      "text_embedding": {
        "type": "dense_vector",
        "index": true,
        "similarity": "cosine"
      }
    }
  }
}


PUT _inference/text_embedding/my-e5-model
{
  "service": "elasticsearch",
  "service_settings": {
    "num_allocations": 1,
    "num_threads": 1,
    "model_id": ".multilingual-e5-small"
  }
}



GET /esre-team-test-chunks/_search
{
  "fields": [
    "text"
  ],
  "knn": {
    "field": "text_embedding",
    "k": 10,
    "num_candidates": 100,
    "query_vector_builder": {
      "text_embedding": {
        "model_id": ".multilingual-e5-small",
        "model_text": "test"
      }
    }
  }
}
```

### Logs (if relevant)

_No response_

### PR DESCRIPTION
**Pull request by kderusso**
Resolves https://github.com/elastic/elasticsearch/issues/111733 

Script to reproduce (slightly expanded from bug report issue): 
```
DELETE esre-team-test-chunks

PUT esre-team-test-chunks
{
  "mappings": {
    "dynamic_templates": [],
    "properties": {
      "metadata": {
        "properties": {
          "esre": {
            "properties": {
              "created_at": {
                "type": "date"
              },
              "embedding_model": {
                "type": "text"
              },
              "file_id": {
                "type": "text"
              },
              "team_id": {
                "type": "text"
              },
              "updated_at": {
                "type": "date"
              }
            }
          },
          "source": {
            "type": "text",
            "fields": {
              "keyword": {
                "type": "keyword",
                "ignore_above": 256
              }
            }
          }
        }
      },
      "text": {
        "type": "text",
        "fields": {
          "ngram": {
            "type": "text"
          }
        }
      },
      "text_embedding": {
        "type": "dense_vector",
        "index": true,
        "similarity": "cosine"
      }
    }
  }
}


PUT _inference/text_embedding/my-e5-model
{
  "service": "elasticsearch",
  "service_settings": {
    "num_allocations": 1,
    "num_threads": 1,
    "model_id": ".multilingual-e5-small"
  }
}

GET /esre-team-test-chunks/_search
{
  "fields": [
    "text"
  ],
  "knn": {
    "field": "text_embedding",
    "k": 10,
    "num_candidates": 100,
    "query_vector_builder": {
      "text_embedding": {
        "model_id": ".multilingual-e5-small",
        "model_text": "test"
      }
    }
  }
}

GET /esre-team-test-chunks/_search
{
  "fields": [
    "text"
  ],
  "knn": {
    "field": "text_embedding",
    "k": 10,
    "num_candidates": 100,
    "query_vector": [ 1.0, 2.0, 3.0 ]
  }
}

```

### CODE REVIEW

File: rest-api-spec/src/yamlRestTest/resources/rest-api-spec/test/search.vectors/40_knn_search.yml
Comment (kderusso) [inline rest-api-spec/src/yamlRestTest/resources/rest-api-spec/test/search.vectors/40_knn_search.yml:1]: Some whitespace changes auto-generated by IDE - "Hide whitespace" to make this more readable.

**Review by benwtrent — APPROVED** [review]

**Review by Samiul-TheSoccerFan — APPROVED** [review]
LGTM

**Comment by kderusso** [comment]
@elasticmachine update branch

**Comment by kderusso** [comment]
## 💚 All backports created successfully

| Status | Branch | Result |
|:------:|:------:|:------|
|✅|8.15|[](https://github.com/elastic/elasticsearch/pull/111763)|

### Questions ?
Please refer to the [Backport tool documentation](https://github.com/sqren/backport)

**Pull request merged**